modules/reports/models/scheduled_reports.php

```diff
@@ -330,144 +330,157 @@ static function get_scheduled_data($schedule_id=false)
 	 * Fetch info on reports to be sent for specific
 	 * period (daily/weekly/monthly)
 	 *
-	 * @param $period_str string: { daily, weekly, monthly }
+	 * @param $period_str string: { schedule }
 	 * @return array
 	 */
 	static function get_period_schedules($period_str)
 	{
-		$id = array();
+		$schedules = array();
 		$send_date = array();
 
 		$db = Database::instance();
-		$sql = "SELECT t1.*,(SELECT value FROM saved_reports_options WHERE report_id = t1.report_id AND name = 'report_timezone') AS timezone FROM scheduled_reports AS t1";
+
+		$sql = <<<'SQL'
+		SELECT sr.*, rp.periodname, opt.value AS timezone
+		FROM scheduled_reports sr
+		INNER JOIN scheduled_report_periods rp ON sr.period_id = rp.id
+		INNER JOIN saved_reports_options opt ON opt.report_id = sr.report_id
```

Comment:
Nitpick: The order of the ON clause comparison is not consistent between these two lines. I prefer putting the joined line first, but either is fine. (Feel free to ignore.)